experiments/train.py

```diff
@@ -0,0 +1,273 @@
+import os
+import time
+import torch
+import torch.utils.data as data
+import torch.distributed as dist
+from copy import deepcopy
+
+from iirc.utils.utils import print_msg
+import lifelong_methods.utils
+from lifelong_methods.buffer.buffer import TaskDataMergedWithBuffer
+from lifelong_methods.utils import transform_labels_names_to_vector
+from experiments import utils, metrics
+
+
+def epoch_train(model, dataloader, config, metadata, gpu=None, rank=0):
+    train_loss = 0
+    train_metrics = {'jaccard_sim': 0., 'modified_jaccard': 0., 'strict_acc': 0., 'recall': 0.}
+    data_len = 0
+    class_names_to_idx = metadata["class_names_to_idx"]
+    num_seen_classes = len(model.seen_classes)
+    model.net.train()
+
+    minibatch_i = 0
+    for minibatch in dataloader:
+        labels_names = list(zip(minibatch[1], minibatch[2]))
+        labels = transform_labels_names_to_vector(
+            labels_names, num_seen_classes, class_names_to_idx
+        )
+
+        if gpu is None:
+            images = minibatch[0].to(config["device"], non_blocking=True)
+            labels = labels.to(config["device"], non_blocking=True)
+        else:
+            images = minibatch[0].to(torch.device(f"cuda:{gpu}"), non_blocking=True)
+            labels = labels.to(torch.device(f"cuda:{gpu}"), non_blocking=True)
+
```

Comment:
If you set config["device"] = torch.device('cpu') or 'cuda', then you don't need to do an if else.

Reply:
this is for the distributed computing when running on multiple gpu (which is not captured by config["device"])